sys/include/net/gnrc/rpl.h

```diff
@@ -347,11 +352,16 @@ kernel_pid_t gnrc_rpl_init(kernel_pid_t if_pid);
  *
  * @param[in] instance_id       Id of the instance
  * @param[in] dodag_id          Id of the DODAG
+ * @param[in] gen_inst_id       Flag indicating whether to generate an instance id.
+ *                              If true, @p instance_id will be ignored
+ * @param[in] local_inst_id     Flag indicating whether a local or global instance id
+ *                              should be generatad
  *
  * @return  Pointer to the new DODAG, on success.
  * @return  NULL, otherwise.
  */
-gnrc_rpl_dodag_t *gnrc_rpl_root_init(uint8_t instance_id, ipv6_addr_t *dodag_id);
+gnrc_rpl_dodag_t *gnrc_rpl_root_init(uint8_t instance_id, ipv6_addr_t *dodag_id, bool gen_inst_id,
```

Comment:
why 2 `bool` for overriding the passed `instance_id`,
I think if `local_inst_id` is `true` then we know that we want to override it.